sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/settings_client.hpp

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Defines the Key Vault Administration client.
+ *
+ */
+
+#pragma once
+#include "settings_client_options.hpp"
+#include <azure/core/context.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <azure/core/response.hpp>
+#include <azure/keyvault/administration/rest_client.hpp>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  class SettingsClient final {
+
+  private:
+    // Using a shared pipeline for a client to share it with LRO (like delete key)
+    Azure::Core::Url m_vaultUrl;
+    std::string m_apiVersion;
+    std::shared_ptr<Azure::Core::Http::_internal::HttpPipeline> m_pipeline;
+
+  public:
+    /**
+     * @brief Destructor.
+     *
+     */
+    virtual ~SettingsClient() = default;
+
+    /**
+     * @brief Construct a new settings client object
+     *
+     * @param vaultUrl The URL address where the client will send the requests to.
+     * @param credential The authentication method to use.
+     * @param options The options to customize the client behavior.
+     */
+    explicit SettingsClient(
+        std::string const& vaultUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+        SettingsClientOptions options = SettingsClientOptions());
+
+    /**
+     * @brief Construct a new settings client object from another settings client.
+     *
+     * @param settingsClient An existing key vault settings client.
+     */
+    explicit SettingsClient(SettingsClient const& settingsClient) = default;
+
+  public:
+    /**
+     * @brief Update a setting
+     *
+     * @param settingName Name of the setting to update
```

Comment:
Be consistent with `.` punctuation.  Some params have them, some don't.  Our standard is to have them.